core/src/main/java/io/grpc/internal/DnsNameResolver.java

```diff
@@ -165,13 +171,18 @@ public void run() {
             savedListener.onError(Status.UNAVAILABLE.withCause(e));
             return;
           }
-          ResolvedServerInfoGroup.Builder servers = ResolvedServerInfoGroup.builder();
+
+          List<InetSocketAddress> inetSocketAddrs = new ArrayList<InetSocketAddress>(
+              inetAddrs.length);
           for (int i = 0; i < inetAddrs.length; i++) {
-            InetAddress inetAddr = inetAddrs[i];
-            servers.add(
-                new ResolvedServerInfo(new InetSocketAddress(inetAddr, port), Attributes.EMPTY));
+            inetSocketAddrs.add(new InetSocketAddress(inetAddrs[i], port));
           }
-          savedListener.onUpdate(Collections.singletonList(servers.build()), Attributes.EMPTY);
+
+          ResolvedServerInfoGroup servers = ResolvedServerInfoGroup.builder()
+              .add(new ResolvedServerInfo(inetSocketAddrs))
```

Comment:
In the case of DNS name resolver, this actually makes way more sense and is semantically correct now. DNS can return multiple addresses for a given hostname, but it is logically still the same destination.

Before, we were creating multiple `ResolvedServerInfo`s which is very confusing and misleading when using DNS name resolver and custom LB - it sort of requires the implementor of LB to have a knowledge about how DNS NR in particular returns addresses and depend LB behaviour on this (making it less portable). Now we create only one object to represent one target host but backed by multiple alternative addresses.